tests/test_async_actor.rs

```diff
@@ -0,0 +1,67 @@
+use std::pin::Pin;
+
+use futures::{future::ready, StreamExt};
+
+use actix::prelude::*;
+use actix::{AsyncHandler, AsyncMessage, TempRef};
+use tokio::time::delay_for;
+use std::time::{Duration, Instant};
+
+#[derive(Debug, Clone, Copy)]
+struct Num(usize);
+
+impl Message for Num {
+    type Result = usize;
+}
+
+impl AsyncMessage for Num {}
+
+struct MyActor {
+    count: usize,
+}
+
+impl Actor for MyActor {
+    type Context = actix::Context<Self>;
+}
+
+impl AsyncHandler<Num> for MyActor {
+    type Result = Pin<Box<dyn Future<Output = <Num as Message>::Result>>>;
+
+    fn handle(
+        mut this: TempRef<Self>,
+        msg: Num,
+        _ctx: TempRef<Self::Context>,
+    ) -> Self::Result {
+        let res = tokio::task::spawn(async move {
+            delay_for(Duration::from_millis(msg.0 as u64)).await;
+            this.as_mut().count += msg.0;
+            this.as_ref().count
+        });
+
+        Box::pin(async move { res.await.unwrap() })
+    }
+}
```

Comment:
Needing a external dependency (`tokio::task::spawn`)  in order to make use of the new trait doesn't look good.

If ActorFuture was spawned as wait on the context we wold be able to implement this handle as:

```Rust
impl Handler<Num> for MyActor {
    type Result = ResponseActFuture<Self, usize>;

    fn handle(&mut self, msg: Num, ctx: &mut Self::Context) -> Self::Result {
        Box::new(
            (delay_for(Duration::from_millis(msg.0 as u64))
                .into_actor(self)
                .map(move |_, this, _| {
                    this.count += msg.0;
                    this.count
                })),
        )
    }
}
```

More on this in the general notes of the PR.

Reply:
There is no need to use an external dependency.

The idea was to show that it can be done. Not always client's async call will be running in the same runtime as actor. (e.g. with async-std one will have to use their runtime ).